app/(main)/league/[leagueId]/entry/all/page.test.tsx

```diff
@@ -65,4 +77,98 @@ describe('Entry Component', () => {
       expect(screen.queryByTestId('global-spinner')).not.toBeInTheDocument();
     });
   });
+
+  it('should display the header with the league name, survivors, and week number, without a past weeks link', async () => {
+    mockUseDataStore.mockReturnValueOnce({ user: { id: '123', leagues: [] } });
+    mockGetGameWeek.mockResolvedValueOnce({ week: 1 });
+    mockGetCurrentUserEntries.mockResolvedValueOnce([
+      {
+        id: '66311a210039f0532044',
+        week: 1,
+      },
+    ]);
+    mockGetCurrentLeague.mockResolvedValueOnce({
+      leagueName: 'GiS League',
+      participants: 47,
+      survivors: 47,
+    });
+
+    render(<Entry params={{ leagueId: '66311a210039f0532044' }} />);
+
+    await waitFor(() => {
+      const entryPageHeader = screen.getByTestId('entry-page-header');
+      const entryPageHeaderToLeaguesLink = screen.getByTestId(
+        'entry-page-header-to-leagues-link',
+      );
+      const entryPageHeaderLeagueName = screen.getByTestId(
+        'entry-page-header-league-name',
+      );
+      const entryPageHeaderLeagueSurvivors =
+        screen.getByTestId('LeagueSurvivors');
+      const entryPageHeaderCurrentWeek = screen.getByTestId(
+        'entry-page-header-current-week',
+      );
+
+      expect(entryPageHeader).toBeInTheDocument();
+      expect(entryPageHeaderToLeaguesLink).toBeInTheDocument();
+      expect(entryPageHeaderToLeaguesLink).toHaveAttribute(
+        'href',
+        '/league/all',
+      );
+      expect(entryPageHeaderLeagueName).toBeInTheDocument();
+      expect(entryPageHeaderLeagueName).toHaveTextContent('GiS League');
+      expect(entryPageHeaderLeagueSurvivors).toBeInTheDocument();
+      expect(entryPageHeaderLeagueSurvivors).toHaveTextContent('Survivors');
+      expect(entryPageHeaderCurrentWeek).toBeInTheDocument();
+      expect(entryPageHeaderCurrentWeek).toHaveTextContent('Week 1');
+    });
+  });
+
+  it('should display the header with the league name, survivors, and week number, with a past weeks link', async () => {
+    mockUseDataStore.mockReturnValueOnce({ user: { id: '123', leagues: [] } });
+    mockGetGameWeek.mockResolvedValueOnce({ week: 2 });
+    mockGetCurrentUserEntries.mockResolvedValueOnce([
+      {
+        id: '66311a210039f0532044',
+        week: 2,
+      },
+    ]);
+    mockGetCurrentLeague.mockResolvedValueOnce({
+      leagueName: 'GiS League',
+      participants: 47,
+      survivors: 47,
```

Comment:
[getCurrentLeague](https://github.com/LetsGetTechnical/gridiron-survivor/blob/6a4e5bf61a63a251aff009b0a3d1b84a7c30cbac/api/apiFunctions.ts#L158) seems to be returning leagueId, leagueName, logo, participants, and survivors. Is this correct?

Reply:
> [getCurrentLeague](https://github.com/LetsGetTechnical/gridiron-survivor/blob/6a4e5bf61a63a251aff009b0a3d1b84a7c30cbac/api/apiFunctions.ts#L158) seems to be returning leagueId, leagueName, logo, participants, and survivors. Is this correct?

@choir27 Yes it is, this is all needed for the information on the page (see screenshots in the PR description). However, this test I do not need the leagueId or logo.